docs/operating-scylla/procedures/maintenance/repair.rst

```diff
@@ -31,8 +31,7 @@ which can run repairs for you.
 To synchronize all data in clusters that have both tablets- and vnodes-based keyspaces, run :doc:`nodetool repair -pr </operating-scylla/nodetool-commands/repair/>` on **all**
 of the nodes in the cluster, and :doc:`nodetool cluster repair </operating-scylla/nodetool-commands/cluster/repair/>` on  **any** of the nodes in the cluster.
 
-.. note:: Run the :doc:`nodetool repair </operating-scylla/nodetool-commands/repair/>` and :doc:`nodetool cluster repair </operating-scylla/nodetool-commands/cluster/repair/>` command regularly. If you delete data frequently, it should be more often than the value of ``gc_grace_seconds`` (by default: 10 days), for example, every week. Use the **nodetool repair -pr** on each node in the cluster, sequentially.
-
+.. note:: Run the :doc:`nodetool repair </operating-scylla/nodetool-commands/repair/>` and :doc:`nodetool cluster repair </operating-scylla/nodetool-commands/cluster/repair/>` command regularly and after every node restart or crash. If you delete data frequently, it should be more often than the value of ``gc_grace_seconds`` (by default: 10 days), for example, every week. Use the **nodetool repair -pr** on each node in the cluster, sequentially.
```

Comment:
Repair is recommended after any even which can cause nodes to miss writes and therefore become inconsistent with other nodes.
A node being down or partitioned is missing writes Restarting a node (be that clean or unclean) also results in said node missing writes.
All that said, I don't think one should rush to do a repair after a restart event. If repair runs on a regular schedule, the next repair will heal the created inconsistency. Also, using QUORUM read/write ensures that so long as only one node is missing data at a time, no inconsistency will be observed by the client.

Reply:
The problem with hard-reboot is that quorum could have been satisfied prior to the reboot but then it's lost, after the write query had been acknowledged.  Reading with quorum consistency after that might return old (or no) data.

Reply:
Right. But this is specific to crashes (hard reboots).

Reply:
doing those after crashes is a bit problematic requirement, i.e. after scylla coredumped, or node kernel panic, who operate the cluster doesn't always know, or have time to do a full cluster repair like suggested here.

and it might go unnoticed for days or weeks.. 

even in testing, we won't be able to stop everything going on, and do this procedure on unplanned crash of scylla or the node.

Reply:
That's the for https://github.com/scylladb/scylladb/issues/17852, in particular with incremental repair, so we can trigger incremental repair for the tablets owned by nodes after restart / disconnect / losing writes (similar to replaying hints today)

Reply:
We probably haven't tested hard / long enough to unearth the extremely rare cases

Reply:
I doubt it.
And I still don't undestand the scenario you're suggesting above.
What does it mean "A crashes, losing X"?
Why would QUORum read from A and C won't see X?
It should be very easy to prove it with a test.py or dtest if you're right.